src/pages/submit/council-decision-veto.tsx

```diff
@@ -0,0 +1,228 @@
+import { SUBMISSION_THRESHOLD_COUNCIL_DECISION_VETO } from '../../constants/proposals'
+import { useAuthContext } from '../../context/AuthProvider'
+import useFormatMessage from '../../hooks/useFormatMessage'
+import useVotingPowerDistribution from '../../hooks/useVotingPowerDistribution'
+import {
+  NewProposalCouncilDecisionVeto,
+  ProposalType,
+  newProposalCouncilDecisionVetoScheme,
+} from '../../types/proposals'
+import locations from '../../utils/locations'
+
+import './council-decision-veto.css'
+import './submit.css'
+
+const initialState: NewProposalCouncilDecisionVeto = {
+  decision_snapshot_id: '',
+  reasons: '',
+  suggestions: '',
+}
+
+const schema = newProposalCouncilDecisionVetoScheme.properties
+
+function getSnapshotIdFromUrl(snapshotUrl: string) {
+  const target = new URL(snapshotUrl)
+  const hash = target.hash.slice(1)
+
+  const proposalMatch = hash.match(/\/proposal\/([^/]+)$/)
+  if (proposalMatch && proposalMatch[1]) {
+    return proposalMatch[1]
+  }
+
+  return hash.split('/').pop() || ''
+}
+
+export default function SubmitCouncilDecisionVeto() {
+  const t = useFormatMessage()
+  const [account, accountState] = useAuthContext()
+  const {
+    handleSubmit,
+    formState: { isDirty, isSubmitting, errors },
+    control,
+    setValue,
+    watch,
+  } = useForm<NewProposalCouncilDecisionVeto>({ defaultValues: initialState, mode: 'onTouched' })
+  const { vpDistribution, isLoadingVpDistribution } = useVotingPowerDistribution(account)
+  const submissionVpNotMet = useMemo(
+    () => !!vpDistribution && vpDistribution.total < Number(SUBMISSION_THRESHOLD_COUNCIL_DECISION_VETO),
+    [vpDistribution]
+  )
+  const [error, setError] = useState('')
+  const navigate = useNavigate()
+
+  const [formDisabled, setFormDisabled] = useState(false)
+  const preventNavigation = useRef(false)
+
+  const setCoAuthors = (addresses?: string[]) => setValue('coAuthors', addresses)
+
+  useEffect(() => {
+    preventNavigation.current = isDirty && !isSubmitting
+  }, [isDirty, isSubmitting])
+
+  const onSubmit: SubmitHandler<NewProposalCouncilDecisionVeto> = async (data) => {
+    setFormDisabled(true)
+
+    const { suggestions, decision_snapshot_id, ...rest } = data
+
+    try {
+      const proposal = await Governance.get().createProposalCouncilDecisionVeto({
+        decision_snapshot_id: getSnapshotIdFromUrl(decision_snapshot_id),
+        suggestions: suggestions || undefined,
+        ...rest,
+      })
+      navigate(locations.proposal(proposal.id, { new: 'true' }), { replace: true })
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (error: any) {
+      setError(error.body?.error || error.message)
+      setFormDisabled(false)
+    }
+  }
+
+  if (accountState.loading) {
+    return <LoadingView />
+  }
+
+  if (!account) {
+    return <LogIn title={t('page.submit_ban_name.title')} description={t('page.submit_ban_name.description')} />
```

Comment:
I guess we don't need new keys for this right?

Reply:
Yes! my bad